crates/net/network/src/state.rs

```diff
@@ -480,6 +480,8 @@ impl<N: NetworkPrimitives> NetworkState<N> {
                 self.on_session_closed(peer)
             }
 
+            let has_received_responses = !received_responses.is_empty();
+
```

Comment:
can we track `no_responses = is_empty` instead?
then we don't have to flip this twice